components/eamxx/src/physics/mam/eamxx_mam_optics_process_interface.cpp

```diff
@@ -1,110 +1,518 @@
+      map_table_name_species_id["pom"] = 3;  // pom:p-organic
+      map_table_name_species_id["bc"] = 5;  // bc :black-c
+      map_table_name_species_id["mom"] = 8;   // mom:m-organic
+
+      for (const auto& item : map_table_name_species_id) {
+       const auto spec_name = item.first;
+       const int species_id = item.second;
+       const auto table_name = "mam4_" + spec_name + "_physical_properties_file";
+       const auto& fname = m_params.get<std::string>(table_name);
+        // read data
+        // need to update table name
+        params_aero.set("Filename", fname);
+        AtmosphereInput refindex_aerosol(params_aero, grid_, host_views_aero,
+                                         layouts_aero);
+        refindex_aerosol.read_variables();
+        refindex_aerosol.finalize();
+        // copy data to device
+        mam_coupling::set_refindex_aerosol(
+            species_id, host_views_aero,
+            specrefndxsw_host,  // complex refractive index for water visible
+            specrefndxlw_host);
+      }  // done ispec
+      // reshape specrefndxsw_host and copy it to device
+      mam4::modal_aer_opt::set_device_specrefindex(
+          aerosol_optics_device_data_.specrefindex_sw, "short_wave",
+          specrefndxsw_host);
+      mam4::modal_aer_opt::set_device_specrefindex(
+          aerosol_optics_device_data_.specrefindex_lw, "long_wave",
+          specrefndxlw_host);
+    }
+  }
+  //FIXME: We are hard-coding the band ordering in RRTMGP.
+  //TODO: We can update optics file using the ordering below (rrtmg_to_rrtmgp_swbands_).
+  // Mapping from old RRTMG sw bands to new band ordering in RRTMGP
+  // rrtmg_swband (old) = 2925, 3625, 4325, 4900, 5650, 6925, 7875, 10450, 14425, 19325,
+    // 25825, 33500, 44000, 1710 ;
+  // rrtmgp_swband (new) = 1710, 2925, 3625, 4325, 4900, 5650, 6925, 7875, 10450, 14425,
+    // 19325, 25825, 33500, 44000 ;
+    // given the rrtmg index return the rrtmgp index
+  std::vector<int> temporal = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 0};
+  auto get_idx_rrtmgp_from_rrtmg_swbands_host =mam_coupling::view_int_1d::HostMirror(temporal.data(),nswbands_);
+  get_idx_rrtmgp_from_rrtmg_swbands_ = mam_coupling::view_int_1d("rrtmg_to_rrtmgp_swbands",nswbands_);
+  Kokkos::deep_copy(get_idx_rrtmgp_from_rrtmg_swbands_,get_idx_rrtmgp_from_rrtmg_swbands_host);
 }
+void MAMOptics::run_impl(const double dt) {
+
+  constexpr Real zero=0.0;
+  constexpr Real one=1.0;
+
+  const auto policy =
+      ekat::ExeSpaceUtils<KT::ExeSpace>::get_default_team_policy(ncol_, nlev_);
+  const auto scan_policy = ekat::ExeSpaceUtils<
+      KT::ExeSpace>::get_thread_range_parallel_scan_team_policy(ncol_, nlev_);
+
+  // preprocess input -- needs a scan for the calculation of atm height
+  Kokkos::parallel_for("preprocess", scan_policy, preprocess_);
+  Kokkos::fence();
+
+  //tau_w_g : aerosol asymmetry parameter * tau * w
+  const auto tau_g_sw = tau_g_sw_;
```

Comment:
I think this should be `tau_ssa_g_sw`, not `tau_g_sw`

Reply:
In the fortran code, the name of this variable is `tau_w_g`; [see here](https://github.com/E3SM-Project/E3SM/blob/7792c63c19659af1ad966f56470cc490302a2bea/components/eam/src/physics/rrtmgp/cam_optics.F90#L503C24-L503C31).  I am okay changing this name from `tau_g_sw` to `tau_ssa_g_sw` or `tau_g_sw` to `tau_w_g`

Reply:
This comment `// tau_w_g : aerosol asymmetry parameter * tau * w` is from [here](https://github.com/eagles-project/e3sm_mam4_refactor/blob/1e8bd59c8282ffe43f9c3428bfa5e5bee4beb83d/components/eam/src/physics/cam/aer_rad_props.F90#L102). Is this comment correct?

Reply:
I don’t care so much about the _sw suffix, it’s the fact that the Fortran version has tau, w, and g to indicate that it’s a product of tau, w, and g, but the c++ version here only has tau and g in the name. So I read that and think this means it’s a product only of tau and g, which is not correct.

Reply:
I changed the name of this variables, from `tau_g_sw` to `tau_ssa_g_sw`. Thanks for checking this.

Reply:
Looks good! Thanks for clarifying this! It will make the next person who looks through this have an easier time understanding what is going on and what these fields represent!